lib/router/router.js

```diff
@@ -109,6 +126,17 @@ p.getInfo = function(req) {
     }
   }
 
+  // forward original image if no operation on image and browser support image format
+  if (routeInfo.imageSteps.length === 0 && (this.options.noop || routeInfo.urlInfo.query.noop)) {
```

Comment:
Did you want to move forward with my suggestions?

It'd be to:

1. Change the behavior of `useOriginal` to not result in storing a cached image when no image steps are provided.
2. Add a new `useOriginal` option to compliment the query, which becomes the default. Would only be the default when no image steps are provided.
3. Add a new `autoFormat` option, allowing you to adhere to the format of the source image, unless the format is not supported by the requesting browser, in which case the existing fallback flow would be used.